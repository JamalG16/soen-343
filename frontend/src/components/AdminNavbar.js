import React from 'react';
import { Icon, Menu, Container } from "semantic-ui-react"

const AdminNavbar = (props) => {
    const { vertical, active, handleTabClick } = props;
    return (
        <Container>
            <Menu vertical={vertical} fixed="top" fluid inverted>
                <Menu.Item 
                    name="overview" 
                    active={active === 'overview'} 
                    onClick={handleTabClick} 
                    style={{ padding: "0 1em", height: "50px"}}>
                        <Icon name="home" />
                    Overview
                </Menu.Item>
                <Menu.Item 
                    name="catalog" 
                    active={active === 'catalog'} 
                    onClick={handleTabClick} 
                    style={{ padding: "0 1em", height: "50px"}}>
                        <Icon name="book" />
                    Catalog
                </Menu.Item>
                <Menu.Item 
                    name="addCatalog" 
                    active={active === 'addCatalog'} 
                    onClick={handleTabClick} 
                    style={{ padding: "0 1em", height: "50px"}}>
                        <Icon name="plus" />
                    Add to catalog
                </Menu.Item>
                <Menu.Item 
                    name="register" 
                    active={active === 'register'} 
                    onClick={handleTabClick}
                    style={{ padding: "0 1em", height: "50px"}}>
                        <Icon name="user plus"/>
                    Register a client
                </Menu.Item>
                <Menu.Item 
                    name="users" 
                    active={active === 'users'} 
                    onClick={handleTabClick}
                    style={{ padding: "0 1em", height: "50px"}}>
                        <Icon name="users" />
                    Active clients
                </Menu.Item>
            </Menu>    
        </Container>
    )
}

export default AdminNavbar;
