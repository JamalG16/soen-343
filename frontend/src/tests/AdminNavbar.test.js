import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import AdminNavbar from '../components/AdminNavbar';

Enzyme.configure({ adapter: new Adapter() });

describe('AdminNavbar', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<AdminNavbar />);
        expect(wrapper).toMatchSnapshot();
    });

    describe('when user clicks tab', () => {
        it('calls the tab click handler', () => {
            const handleTabClickMock = jest.fn();
            const wrapper = shallow(
                <AdminNavbar handleTabClick={handleTabClickMock}/>
            );

            wrapper.find('[name="overview"]').simulate('click');
            expect(handleTabClickMock).toHaveBeenCalledTimes(1);
        });
    });
});