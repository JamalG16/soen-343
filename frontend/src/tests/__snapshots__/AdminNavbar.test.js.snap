// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminNavbar renders correctly 1`] = `
<Container>
  <Menu
    fixed="top"
    fluid={true}
    inverted={true}
  >
    <MenuItem
      active={false}
      name="overview"
      style={
        Object {
          "height": "50px",
          "padding": "0 1em",
        }
      }
    >
      <Icon
        as="i"
        name="home"
      />
      Overview
    </MenuItem>
    <MenuItem
      active={false}
      name="register"
      style={
        Object {
          "height": "50px",
          "padding": "0 1em",
        }
      }
    >
      <Icon
        as="i"
        name="user plus"
      />
      Register a client
    </MenuItem>
    <MenuItem
      active={false}
      name="users"
      style={
        Object {
          "height": "50px",
          "padding": "0 1em",
        }
      }
    >
      <Icon
        as="i"
        name="users"
      />
      Active clients
    </MenuItem>
  </Menu>
</Container>
`;
